File: server/public/model/ai.go
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package model

// AIRequest represents a request to call the AI plugin
type AIRequest struct {
	Data           map[string]any `json:"data"`
	ResponseSchema []byte         `json:"response_schema,omitempty"`
}

File: webapp/channels/src/packages/mattermost-redux/src/reducers/entities/ai.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {combineReducers} from 'redux';

import type {MMReduxAction} from 'mattermost-redux/action_types';

import {AITypes} from '../../action_types';
import type {AIAgent} from '../../actions/ai';

export interface AIState {
    agents: AIAgent[];
}

function agents(state: AIAgent[] = [], action: MMReduxAction): AIAgent[] {
    switch (action.type) {
    case AITypes.RECEIVED_AI_AGENTS:
        return action.data;
    default:
        return state;
    }
}

export default combineReducers({
    agents,
});

